qemu/tests/numa_cpu.py

```diff
@@ -0,0 +1,139 @@
+            if vm_arch in ('x86_64', 'i386'):
+                socket_min, socket_max = _get_boundary(socket, vcpu_sockets, socket_weight)
+                die_min, die_max = _get_boundary(die, vcpu_dies, die_weight)
+                core_min, core_max = _get_boundary(core, vcpu_cores, core_weight)
+                thread_min, thread_max = _get_boundary(thread, vcpu_threads, thread_weight)
+                cpu_min = socket_min + die_min + core_min + thread_min
+                cpu_max = socket_max + die_max + core_max + thread_max
+            elif vm_arch in ('ppc64', 'ppc64le'):
+                cpu_min = int(core)
+                cpu_max = int(core) + vcpu_threads - 1
+
+            if nodeid in tmp.keys():
+                tmp[nodeid] += range(cpu_min, cpu_max+1, 1)
+            else:
+                tmp[nodeid] = range(cpu_min, cpu_max+1, 1)
+        for item in sorted(tmp.items(), key=lambda item: item[0]):
+            expected_cpus.append(set(sorted(item[1])))
+        return expected_cpus
+
+    def get_hotpluggable_cpus():
+        """
+        Parse output of "query-hotpluggable-cpus", get the cpus that bind to
+        special node.
+        """
+        error_context.context("Get the hotpluggable cpus", logging.info)
+        hotpluggable_cpus = []
+        tmp = {}
+        out = vm.monitor.info("hotpluggable-cpus")
+        for vcpu_info in out:
+            vcpus_count = vcpu_info["vcpus-count"]
+            vcpu_info = vcpu_info["props"]
+            nodeid = vcpu_info.get("node-id")
+            if nodeid is not None:
+                socketid = vcpu_info.get("socket-id")
+                dieid = vcpu_info.get("die-id")
+                coreid = vcpu_info.get("core-id")
+                threadid = vcpu_info.get("thread-id")
+                if vm_arch in ('x86_64', 'i386'):
+                    cpu_index = [socketid * socket_weight + dieid * die_weight +
+                                 coreid * core_weight + threadid * thread_weight]
+                elif vm_arch in ('ppc64', 'ppc64le'):
+                    cpu_index = range(coreid, coreid + vcpus_count)
+                if nodeid in tmp.keys():
+                    tmp[nodeid] += cpu_index
+                else:
+                    tmp[nodeid] = cpu_index
+        for item in sorted(tmp.items(), key=lambda item: item[0]):
+            hotpluggable_cpus.append(set(sorted(item[1])))
+        return hotpluggable_cpus
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    os_type = params["os_type"]
+    vm_arch = params["vm_arch_name"]
+
+    vcpu_sockets = params.get_numeric('vcpu_sockets')
+    vcpu_dies = params.get('vcpu_dies')
+    if vcpu_dies == 'INVALID':
+        vcpu_dies = 1
+    else:
+        vcpu_dies = int(vcpu_dies)
+    vcpu_cores = params.get_numeric('vcpu_cores')
+    vcpu_threads = params.get_numeric('vcpu_threads')
+
+    socket_weight = vcpu_dies * vcpu_cores * vcpu_threads
```

Comment:
They are not used by PPC, so I would suggest moving them to x86 specific part, and then `if vcpu_dies == 'INVALID':` can be removed.

Others LGTM.

Reply:
Good point, updated, thanks!